CMakeLists.txt

```
@@ -16,6 +16,14 @@ option(CMAKE_RUN_CLANG_TIDY "Run clang-tidy" OFF)
 option(LIBOCPP_BUILD_EXAMPLES "Build charge_point and central_system binaries." OFF)
 option(OCPP_INSTALL "Install the library (shared data might be installed anyway)" ${EVC_MAIN_PROJECT})
 
+if (${CMAKE_CXX_STANDARD} LESS_EQUAL 17) 
```

```suggestion
if (${CXX_STANDARD} LESS 17) 
```

1. `CMAKE_CXX_STANDARD` is only the default value for `CXX_STANDARD`. See https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html
2. `LESS` because `std::filesystem` exists since C++17.

---

We could also skip the C++ version detection here completely and rely on the user the provide the option when this is desired, like I did in https://github.com/EVerest/liblog/pull/13

---

Yeah, actually I like this approach more. As this helps that the build works out of the box. And it discourages to use `boost::filesystem` on newer C++-versions. So it's explicit that this serves only as a backwards-compatibility.

---

On my system CMAKE_CXX_STANDARD delivered 17 although std::filesystem was not present. So I am pruposing to bind it to the compiler version:
Some interesting note on that, i found here: ["C++17 mode is the default since GCC 11; it can be explicitly selected with the -std=c++17 command-line flag, or -std=gnu++17 to enable GNU extensions as well. Some C++17 features are available since GCC 5, but support was experimental and the ABI of C++17 features was not stable until GCC 9. "](https://gcc.gnu.org/projects/cxx-status.html#cxx17)

---

Hmm, interesting. Did you [explicitly set the compiler for `cmake`](https://stackoverflow.com/questions/45933732/how-to-specify-a-compiler-in-cmake)?

Otherwise perhaps the `CXX`-env points to `gcc` and `cmake` then uses the default for `gcc` (which is C++17 since GCC11, as you pointed out).

Which compiler and version are you using? (If you can't name it publicly, perhaps via e-mail to me?)

---

I tried CXX_STANDARD but this was empty.

---

Given that there seem to be issues with the automatic detection of this I would propose using a cmake flag like
LIBOCPP_USE_BOOST_FILESYSTEM that you have to explicitly set (like in the aforementioned PR in liblog)

---

Ok, fine. I suggest then to name it without the `LIBOCPP`-prefix. Otherwise you'll need to set multiple flags for the same feature. 

To be future-proof: `USE_NON_STD_FILESYSTEM`. It was suggested within Pionix to consider an alternative to `boost::filesystem`, e.g. https://github.com/gulrak/filesystem. Given that we're currently seeing multiple issues with it. See related
- https://github.com/EVerest/liblog/pull/13
- https://github.com/EVerest/everest-core/pull/348